### ISSUE TITLE
BUG: Change in index of empty dataframes in `mode` operation

### ISSUE THREAD

**Issue by galipremsagar** [pandas-dev/pandas#43336]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example
Previously pandas used to retain the index in an empty dataframe:
```python
>>> df = pd.DataFrame({'a':['a', 'b', 'c']}, index=['a', 'b', 'c'])
>>> df.mode()
   a
0  a
1  b
2  c
>>> df.mode(numeric_only=True)
Empty DataFrame
Columns: []
Index: [a, b, c]
>>> pd.__version__
'1.2.5'

```

Currently the index is not being retained if the dataframe is empty:

```python
>>> import pandas as pd
>>> df = pd.DataFrame({'a':['a', 'b', 'c']}, index=['a', 'b', 'c'])
>>> df.mode()
   a
0  a
1  b
2  c
>>> df.mode(numeric_only=True)
Empty DataFrame
Columns: []
Index: []
>>> pd.__version__
```

Is this a conscious change, if yes can someone point me to the docs where I can learn more about this change?

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 5f648bf1706dd75a9ca0d29f26eadfbb595fe52b
python           : 3.8.10.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-76-generic
Version          : #86-Ubuntu SMP Fri Jan 17 17:24:28 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.3.2
numpy            : 1.21.2
pytz             : 2021.1
dateutil         : 2.8.2
pip              : 21.2.4
setuptools       : 57.4.0
Cython           : 0.29.24
pytest           : 6.2.5
hypothesis       : 6.17.4
sphinx           : 4.1.2
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.1
IPython          : 7.27.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : None
fsspec           : 2021.08.1
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 5.0.0
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : 0.53.1

</details>

**Comment by simonjayhawkins** [comment]
Thanks @galipremsagar for the report.

The new behavior looks potentially correct to me. (It maybe should raise instead @jbrockmendel?)

The number of rows in the result of a mode operation (with the default axis) corresponds to the highest number of modes in the results over all the columns. 

For the first example, there are 3 modes in the only column present, hence 3 rows in the result.

In the second example there are no columns to evaluate (since `numeric_only=True`) and hence no modes so I would expect the result to have no rows.

> Is this a conscious change, if yes can someone point me to the docs where I can learn more about this change?

I do a bisect shortly to get more context.

**Comment by simonjayhawkins** [comment]
> > Is this a conscious change, if yes can someone point me to the docs where I can learn more about this change?
> 
> I do a bisect shortly to get more context.

change looks intentional to fix #33321

[508d0d204ea5151cd64756425611a2c26e5365b3] BUG: Ensure series/frame mode() keeps int index (#38732)

**Comment by galipremsagar** [comment]
@simonjayhawkins Thanks for clarifying, closing this issue.
